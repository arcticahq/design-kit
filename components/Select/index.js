import React from "react"
import { withFormsy, propTypes } from "formsy-react"

import styles from "./index.module.css"

class Select extends React.Component {
  state = {
    blurred: false
  }

  render() {
    const {
      name,
      title,
      showError,
      getErrorMessage,
      showRequired,
      isPristine,
      getValue,
      setValue,
      options,
      required
    } = this.props
    const { blurred } = this.state

    const className = `form-group ${showError() ? "error" : ""}`
    const requiredClass = showRequired() && !isPristine() ? `required` : ``
    const errorMessage = blurred ? getErrorMessage() : ``

    const selectOptions = options.map(option => (
      <option key={option.title + option.value} value={option.value}>
        {option.title}
      </option>
    ))

    return (
      <div className={className}>
        <label className={requiredClass} htmlFor={name}>
          {title} {required && <span className="required">*</span>}
          <select
            className={styles.select}
            name={name}
            id={name}
            onChange={e => setValue(e.currentTarget.value)}
            value={getValue()}
          >
            {selectOptions}
          </select>
        </label>
        <span className="validation-error">{errorMessage}</span>
      </div>
    )
  }
}

Select.propTypes = {
  ...propTypes
}

export default Select
export const FormsySelect = withFormsy(Select)
