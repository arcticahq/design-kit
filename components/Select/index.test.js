import React from "react"
import { shallow } from "enzyme"
import Select from "."

describe(`Select`, () => {
  let props
  let mountedComponent
  const wrapper = () => {
    if (!mountedComponent) {
      mountedComponent = shallow(<Select {...props} />)
    }
    return mountedComponent
  }

  beforeEach(() => {
    props = {
      name: `type`,
      title: `Select`,
      options: [
        { title: `Freelance`, value: `freelance` },
        { title: `Contact Opportunity`, value: `contract` },
        { title: `Other`, value: `other` }
      ],
      required: true,
      showRequired: () => true,
      isPristine: () => true,
      showError: () => jest.fn(),
      getErrorMessage: () => jest.fn(),
      setValue: () => jest.fn(),
      getValue: () => jest.fn()
    }
    mountedComponent = undefined
  })

  it(`renders an Select correctly`, () => {
    expect(wrapper()).toMatchSnapshot()
  })
})
