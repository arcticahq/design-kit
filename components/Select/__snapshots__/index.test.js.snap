// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders an Select correctly 1`] = `
<div
  className="form-group error"
>
  <label
    className=""
    htmlFor="type"
  >
    Select
     
    <span
      className="required"
    >
      *
    </span>
    <select
      className="select"
      id="type"
      name="type"
      onChange={[Function]}
      value={[MockFunction]}
    >
      <option
        key="Freelancefreelance"
        value="freelance"
      >
        Freelance
      </option>
      <option
        key="Contact Opportunitycontract"
        value="contract"
      >
        Contact Opportunity
      </option>
      <option
        key="Otherother"
        value="other"
      >
        Other
      </option>
    </select>
  </label>
  <span
    className="validation-error"
  />
</div>
`;
